source/adapters/hip/usm.cpp

```diff
@@ -170,19 +169,27 @@ urUSMGetMemAllocInfo(ur_context_handle_t hContext, const void *pMem,
       // Direct usage of the function, instead of UR_CHECK_ERROR, so we can get
       // the line offset.
       checkErrorUR(Ret, __func__, __LINE__ - 5, __FILE__);
-      Value = hipPointerAttributeType.isManaged;
-      if (Value) {
-        // pointer to managed memory
-        return ReturnValue(UR_USM_TYPE_SHARED);
+      // ROCm 6.0.0 introduces hipMemoryTypeUnregistered in the hipMemoryType
+      // enum to mark unregistered allocations (i.e., via system allocators).
+#if HIP_VERSION_MAJOR >= 6
+      if (hipPointerAttributeType.type == hipMemoryTypeUnregistered) {
+        // pointer not known to the HIP subsystem
+        return ReturnValue(UR_USM_TYPE_UNKNOWN);
       }
-      UR_CHECK_ERROR(hipPointerGetAttributes(&hipPointerAttributeType, pMem));
```

Comment:
This line was removed as I couldn't figure out the need for re-querying if we've already validated the return value of the first `hipError_t Ret = hipPointerGetAttributes(&hipPointerAttributeType, pMem);` call. The first call does not throw as it is used to check the reutrn value but we call `checkErrorUR` on that return value a few lines later which will still throw on non-success.

Not sure if I am missing something but I don't think it is needed.

Reply:
That looks OK to me